api/base/filters.py

```diff
@@ -66,6 +69,33 @@ def filter_queryset(self, request, queryset, view):
             return queryset.sort(*ordering)
         return queryset
 
+    def check_serializer_fields(self, queryset, query_params, view, ordering):
+        serializer_class = view.serializer_class
+        if not queryset:
+            return []
+        sorting_params = []
+        for field in query_params.getlist(self.ordering_param):
+            ordering_sign = ''
+            # If the field is already a source, it will be returned in the ordering list
+            if field in ordering:
+                sorting_params.append(field)
+                continue
+            # The field may have a '-' before the name for descending order. This needs to be preserved
+            if field[0] == '-':
+                ordering_sign = '-'
+                field = field.lstrip('-')
+            if field in serializer_class._declared_fields:
+                # Checking if the field can be sorted on
+                source_field = serializer_class._declared_fields[field]
+                source_field_name = source_field.source
```

Comment:
Another insight from inspecting DRF's `OrderingFilter.get_default_valid_fields`, a good reminder that the source may not be one level, but could be multiple levels deep - they are doing `field.source.replace('.', '__') `
```python
return [
            (field.source.replace('.', '__') or field_name, field.label)
            for field_name, field in serializer_class(context=context).fields.items()
            if not getattr(field, 'write_only', False) and not field.source == '*'
        ]
```

Also, they're excluding write_only fields and preventing the source from being '*'.